prow/plugins/label/label.go

```diff
@@ -218,6 +227,17 @@ func handle(gc githubClient, log *logrus.Entry, config plugins.Label, e *github.
 		if err := gc.AddLabel(org, repo, e.Number, labelToAdd); err != nil {
 			log.WithError(err).Errorf("GitHub failed to add the following label: %s", labelToAdd)
 		}
+
+		// Assign configured users if present
+		for _, restrictedLabel := range restrictedLabels {
+			for _, assignOn := range restrictedLabel.AssignOn {
```

Comment:
This code assumes that the `assign_on` label always gets added by the `label` plugin which might or might not be true. IMHO it should instead react to the label adding event.

Reply:
I'm not sure I follow what you mean by "label adding event", can you elaborate?

Reply:
https://docs.github.com/en/developers/webhooks-and-events/events/github-event-types#issuesevent

Reply:
The intention, as stated in https://github.com/kubernetes/test-infra/issues/24394, was to use the `label` plugin itself. I do, however, agree that it would be more robust if _any_ addition of a label results in this functionality. Would this mean creating a separate plugin to handle this though?

Reply:
The intention as stated in that issue is to use the label plugin to do the assignment, it doesn't really say anything about where the event comes from. Requiring that the label plugin adds the label to get the trigger seems unintuitive and prone to be an issue sooner or later

Reply:
I think I get what you are saying now. So rather than having this new functionality exist in the `labelsToAdd` loop we would just check for the existence of the label and assign the users outside of that loop.